### ISSUE TITLE
Custom detekt rules are not working

### ISSUE THREAD

**Issue by volsahin** [detekt/detekt#2663]
## Observed Behavior
Detekt itself works fine but custom-rules are not reported after executing detekt task.

## Steps to Reproduce
* Create a module detekt-custom-rules
* Create a rule inside this module
```
class SampleRule() : Rule() {

    override val issue = Issue(
        javaClass.simpleName,
        Severity.Maintainability,
        "Sample rule",
        Debt(hours = 1)
    )

    override fun visitKtFile(file: PsiFile?) {
        super.visitKtFile(file)
        report(CodeSmell(issue, Entity.from(file!!),"Reporting..."))
    }
}
```
* Create a provider inside this module
```
class CustomRulesetProvider : RuleSetProvider {
    override val ruleSetId: String = "custom-ruleset"

    override fun instance(config: Config): RuleSet = RuleSet(
        ruleSetId, listOf(SampleRule())
    )
}
```
* Add provider package name to file:
`resources/META-INF/services/io.gitlab.arturbosch.detekt.api.RuleSetProvider`
* Add custom properties to detekt.yml
```
config:
  validation: true
  excludes: "custom-ruleset,.*>.*"

custom-ruleset:
  active: true
  SampleRule:
    active: true
```
* And add detekt-custom-rules module to app module
```
detekt "io.gitlab.arturbosch.detekt:detekt-cli:1.8.0"
detekt project(path: ':detekt-custom-rules', configuration: 'default')
```

## Observations
* I follow the steps on https://detekt.github.io/detekt/extensions.html#pitfalls and connect detekt task with module build with
`tasks.withType<Detekt> { dependsOn(":my-rules:assemble") }`
* Version of detekt used: 1.6.0, 1.8.0

**Comment by arturbosch** [comment]
Hey,
try changing this line:
`detekt project(path: ':detekt-custom-rules', configuration: 'default')`
to:
`detektPlugins project(path: ':detekt-custom-rules', configuration: 'default')`
and it should work.
Notice that you normally don't need to specify the cli dependency.
If you want to use another underlying detekt version, try using the `toolVersion` property inside the `detekt` gradle extension class.

**Comment by volsahin** [comment]
Hey, thanks for replying. Actually i tried with using detektPlugins but it didn't work either.

**Comment by volsahin** [comment]
Ok after a few days i have just seen what is the problem. I created my detekt-custom-rules module as an android module. 

So changing these lines
```
apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-android-extensions'
```
with this one
`apply plugin: 'kotlin'`

solved my problem.
Maybe we can add a line to documentation to avoid potential misunderstandings.

And also after converting detekt to detektPlugins, i could able to delete cli dependency.
Thanks for information @arturbosch

**Comment by arturbosch** [comment]
@volsahin 
Would you be up to open a PR with the extra documentation? Maybe an entry under pitfalls would be a good place - https://detekt.github.io/detekt/extensions.html#pitfalls , thanks.

**Comment by schalkms** [comment]
Glad that you could solve the problem.
@volsahin Do you mind to help us out and improve the documentation for other users as well?
May I ask you to write a few lines about the circumstance? The perfect would be [This way the probl](here).
This way the problem description is persisted and not only hidden in a searchable issue.

**Comment by schalkms** [comment]
@arturbosch you were a few seconds faster 😁

**Comment by volsahin** [comment]
Ok i will surely do :) Thanks @arturbosch @schalkms
